modules/srv6/gr_srv6.h

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025 Olivier Gournet
+
+#ifndef _GR_DATAPATH_SRV6
+#define _GR_DATAPATH_SRV6
+
+#include "gr_srv6_api.h"
+
+#include <rte_ip6.h>
+
+#include <endian.h>
+#include <netinet/ip6.h>
+
+//
+// srv6 pkt header definition
+//
+#define IP6_ROUTING_HEADER_TYPE_SRV6 4
+
+struct ip6_sr_rthdr {
+	uint8_t nxt; /* next header */
+	uint8_t len; /* length in units of 8 octets */
+	uint8_t type; /* routing type */
+	uint8_t segleft; /* segments left */
+	uint8_t last_entry;
+	uint8_t flags;
+	rte_be16_t tag;
+	struct rte_ipv6_addr segments[];
+} __rte_packed;
```

Comment:
Could you reuse the routing header extension that is already defined in dpdk?

https://github.com/DPDK/dpdk/blob/fd233ad17e5ffa42d50f1625165a7fce3f1cbc5f/lib/net/rte_ip6.h#L511-L526